lib/Recur/RRuleIterator.php

```diff
@@ -408,13 +451,13 @@ protected function nextMonthly(): void
             // occur to the next month. We Must skip these invalid
             // entries.
             if ($currentDayOfMonth < 29) {
-                $this->currentDate = $this->currentDate->modify('+'.$this->interval.' months');
+                $this->advanceTheDate('+'.$this->interval.' months');
             } else {
                 $increase = 0;
                 do {
                     ++$increase;
                     $tempDate = clone $this->currentDate;
-                    $tempDate = $tempDate->modify('+ '.($this->interval * $increase).' months');
+                    $tempDate = $tempDate->modify('+ '.($this->interval * $increase).' months '.$this->startDate->format('H:i:s'));
```

Comment:
`advanceTheDate` directly adjusts `$this->currentDate`

But this code adjusts `$tempDate` which is a separate clone of `currentDate` that is used in a loop, to eventually become a new `currentDate` after possibly looping over some months.

But I will try to put the `$this->startDate->format('H:i:s')` snippet into a private method to avoid repeating that "magic" in many places.

Reply:
> But I will try to put the `$this->startDate->format('H:i:s')` snippet into a protected method to avoid repeating that "magic" in many places.

protected method `startTime()` added and used.